wpimath/src/main/java/edu/wpi/first/math/kinematics/ChassisSpeeds.java

```diff
@@ -77,6 +78,28 @@ public ChassisSpeeds(
     this(vx.in(MetersPerSecond), vy.in(MetersPerSecond), omega.in(RadiansPerSecond));
   }
 
+  /**
+   * Creates a Twist2d from ChassisSpeeds.
+   *
+   * @param dtSeconds The duration of the timestep.
+   * @return Twist2d.
+   */
+  public Twist2d toTwist2d(double dtSeconds) {
+    // Construct the desired pose after a timestep, relative to the current pose. The desired pose
+    // has decoupled translation and rotation.
+    var desiredDeltaPose =
+        new Pose2d(
+            vxMetersPerSecond * dtSeconds,
+            vyMetersPerSecond * dtSeconds,
+            new Rotation2d(omegaRadiansPerSecond * dtSeconds));
+
+    // Find the chassis translation/rotation deltas in the robot frame that move the robot from its
+    // current pose to the desired pose
+    var twist = Pose2d.kZero.log(desiredDeltaPose);
```

Comment:
Why isn't the implementation just:
```java
return new Twist2d(vxMetersPerSecond * dtSeconds,
                                    vyMetersPerSecond * dtSeconds,
                                    omegaRadiansPerSeconds * dtSeconds);
}
```

Reply:
I was mainly basing it off of the conversion done for discretize, I will fix that. I imagine the same should be done for C++?

Reply:
Yes.